packages/user/src/model/invitations/resolver.ts

```diff
@@ -0,0 +1,363 @@
+import mercurius from "mercurius";
+import { createNewSession } from "supertokens-node/recipe/session";
+import {
+  emailPasswordSignUp,
+  getUsersByEmail,
+} from "supertokens-node/recipe/thirdpartyemailpassword";
+import UserRoles from "supertokens-node/recipe/userroles";
+
+import Service from "./service";
+import computeInvitationExpiresAt from "../../lib/computeInvitationExpiresAt";
+import formatDate from "../../lib/formatDate";
+import isInvitationValid from "../../lib/isInvitationValid";
+import sendInvitation from "../../lib/sendInvitation";
+import validateEmail from "../../validator/email";
+import validatePassword from "../../validator/password";
+
+import type {
+  Invitation,
+  InvitationCreateInput,
+  InvitationUpdateInput,
+} from "../../types/invitation";
+import type { FilterInput, SortInput } from "@dzangolab/fastify-slonik";
+import type { MercuriusContext } from "mercurius";
+import type { QueryResultRow } from "slonik";
+
+const Mutation = {
+  acceptInvitation: async (
+    parent: unknown,
+    arguments_: {
+      data: {
+        email: string;
+        password: string;
+      };
+      token: string;
+    },
+    context: MercuriusContext
+  ) => {
+    const { app, config, database, dbSchema, reply } = context;
+
+    const { token, data } = arguments_;
+
+    try {
+      const { email, password } = data;
+
+      //  check if the email is valid
+      const emailResult = validateEmail(email, config);
+
+      if (!emailResult.success && emailResult.message) {
+        const mercuriusError = new mercurius.ErrorWithProps(
+          emailResult.message
+        );
+
+        return mercuriusError;
+      }
+
+      // password strength validation
+      const passwordStrength = validatePassword(password, config);
+
+      if (!passwordStrength.success && passwordStrength.message) {
+        const mercuriusError = new mercurius.ErrorWithProps(
+          passwordStrength.message
+        );
+
+        return mercuriusError;
+      }
+
+      const service = new Service<
+        Invitation & QueryResultRow,
+        InvitationCreateInput,
+        InvitationUpdateInput
+      >(config, database, dbSchema);
+
+      const invitation = await service.findByToken(token);
+
+      // validate the invitation
+      if (!invitation || !isInvitationValid(invitation)) {
+        const mercuriusError = new mercurius.ErrorWithProps(
+          "Invitation is invalid or has expired"
+        );
+
+        return mercuriusError;
+      }
+
+      // compare the FieldInput email to the invitation email
+      if (invitation.email != email) {
+        const mercuriusError = new mercurius.ErrorWithProps(
+          "Email do not match with the invitation"
+        );
+
+        return mercuriusError;
+      }
+
+      // signup
+      const signUpResponse = await emailPasswordSignUp(email, password);
+
+      if (!(signUpResponse.status === "OK")) {
```

Comment:
can't it be just 

` if (signUpResponse.status !== "OK") {`  ?

Reply:
yes, it can be.